agent/base

```diff
@@ -12,24 +12,20 @@ set -o pipefail
 # very first thing to do is figure out which pbench we are
 if [[ -z "$pbench_run" ]]; then
     pbench_run=$(pbench-config pbench_run pbench-agent)
-    if [[ -z "$pbench_run" ]]; then
-        pbench_run=/var/lib/pbench-agent
-    fi
```

Comment:
This is unrelated to the change we'd discussed. In fact, Python has similar code, where the `PbenchAgentConfig` constructor supplies a hardcoded default if the `pbench_run` setting isn't defined in the config file. If we're working to improve consistency, we want this to stay.

And, now that I look at the Python more carefully ... `PbenchAgentConfig` goes on to `self.pbench_tmp.mkdir(parents=True, exist_ok=True)` so any Python agent command using `PbenchAgentConfig` is guaranteed to have the full `/var/lib/pbench_agent/tmp` created if it didn't already exist.

The `PbenchAgentConfig` however doesn't do anything about a user-supplied `pbench_run` override; the real difference was that, in *that* case, I believe the Python command code we looked at before would fail rather than creating it.

Reply:
Do we want to pass a hardcoded `pbench_run` path? I feel providing a hardcoded path might create problem in the future.